src/testing/flow_solver_cases/periodic_cube_flow.cpp

```diff
@@ -210,9 +210,41 @@ double PeriodicTurbulence<dim, nstate>::compute_kinetic_energy(DGBase<dim, doubl
     return integrate_over_domain(dg, IntegratedQuantitiesEnum::kinetic_energy);
 }
 
+//=========================================================
+// PERIODIC 1D DOMAIN FOR TIME REFINEMENT STUDY
+//=========================================================
+
+template <int dim, int nstate>
+Periodic1DFlow<dim, nstate>::Periodic1DFlow(const PHiLiP::Parameters::AllParameters *const parameters_input)
+        : PeriodicCubeFlow<dim, nstate>(parameters_input)
+{
+}
+
+template <int dim, int nstate>
+void Periodic1DFlow<dim, nstate>::compute_unsteady_data_and_write_to_table(
+       const unsigned int current_iteration,
+        const double current_time,
+        const std::shared_ptr <DGBase<dim, double>>/* dg */,
+        const std::shared_ptr <dealii::TableHandler> /* unsteady_data_table */) const
+{
+    double dt = this->all_param.ode_solver_param.initial_time_step;
+    int output_solution_every_n_iterations = (int) (this->all_param.ode_solver_param.output_solution_every_dt_time_intervals/dt);
+ 
+    if ((current_iteration % output_solution_every_n_iterations) == 0){
+        this->pcout << "    Iter: " << current_iteration
+                    << "    Time: " << current_time
+                    << std::endl;
+    }
```

Comment:
Instead of creating this class for the sole purpose of printing iteration and time, this should probably be done in the ODE solver `step_in_time` function. 

After line 55 here: https://github.com/dougshidong/PHiLiP/blob/1bed06d44bd9bcae1733c92efbca7b313262d81d/src/ode_solver/explicit_ode_solver.cpp#L55 

You could add:
`if ((this->ode_param.ode_output) == Parameters::OutputEnum::verbose &&
            (this->current_iteration%ode_param.print_iteration_modulo) == 0 ) {
           this->pcout << "    Iter: " << this->current_iteration
                                << "    Time: " << this->current_time
                                << std::endl;
}`

and use those parameters, or what ever makes most sense, in your `.prm` file.

Reply:
I could do that for flow_solver cases. However, that would make cases which use `advance_solution_time` print the current iteration twice due to [this](https://github.com/dougshidong/PHiLiP/blob/1bed06d44bd9bcae1733c92efbca7b313262d81d/src/ode_solver/ode_solver_base.cpp#L229) in `ode_solver_base.cpp`. In theory, I could add another statement to check whether it's a flow solver case, but I still feel like that is repeated functionality between `advance_solution_time` and 'step_in_time`. Also, I'll soon be adding a 1D periodic time refinement test which needs to track unsteady data. Therefore, while this class doesn't do much now, it is useful to keep here for future generality.

Reply:
Good call, maybe just rename this class as `AdvectionPeriodic` then since its confusing having `PeriodicCubeFlow` and a `Periodic1DFlow` when cube includes them all

Reply:
And further move the `AdvectionPeriodic` class into to a separate file `flow_solver_cases/advection_periodic.cpp/.h`

Reply:
I am renaming to `Periodic1DUnsteady`, as this same flow solver setup will be used with Burgers' for a future time refinement study. I am also moving into separate files, `periodic_1D_unsteady.<cpp/h>`.